src/main/java/org/jvnet/hudson/plugins/platformlabeler/NodeLabelCache.java

```diff
@@ -57,7 +59,25 @@ public class NodeLabelCache extends ComputerListener {
     private static final transient Logger LOGGER = Logger.getLogger("org.jvnet.hudson.plugins.platformlabeler");
 
     /**
-     * When a computer comes online, probe it for its platform labels.
+     * When a computer is about to come online, probe it for its platform labels.
+     * Typically for ephemeral agent that connect and disconnect frequently
+     *
+     * @param computer agent whose labels will be cached
+     * @param channel This is the channel object to talk to the agent
+     * @param root The directory where this agent stores files. The same as Node.getRootPath(), except that method returns null until the agent is connected. So this parameter is passed explicitly instead.
+     * @param ignored TaskListener that is ignored
+     * @throws java.io.IOException on IO error
+     * @throws java.lang.InterruptedException on thread interrupt
+     */
+    @Override
+    public final void preOnline(final Computer computer, Channel channel, FilePath root, final TaskListener ignored)
+            throws IOException, InterruptedException {
+        cacheAndRefreshModel(computer, channel);
```

Comment:
Should we maybe catch any exception here? When we let the exception pass it will prevent that the agent comes online at all.

Reply:
That's a good point, I was thinking the same. I'm not sure when/how platformlabeler can raise exception but It's probably safer to catch error